### PR TITLE
jruuuu/APPEALS-8832: Update rails to 5.2.8  to update  Actionpak, ActionView, Activerecord, and ActiveStorage 

### PR DESCRIPTION
**Pull request by Jruuuu**
upgraded rails to 5.2.8 in order to update Actionpak, ActionView, Activerecord and Active Storage to resolve security vulnerabilities reported by bundle auditor. please refer [CircleCI Repo Results SpreadSheet](https://boozallen.sharepoint.com/:x:/r/teams/VABID/appeals/_layouts/15/doc2.aspx?sourcedoc=%7BD6A10F90-3C85-466C-BC98-CDBA4422119E%7D&file=CircleCi%20Repo%20Results.xlsx&action=default&mobileredirect=true&cid=d71ba0c4-8ea9-4985-90c5-c8a0c0d7394f) 

Resolves # [APPEALS-13649](https://vajira.max.gov/browse/APPEALS-13649) | [APPEALS-13652](https://vajira.max.gov/browse/APPEALS-13652) |  [APPEALS-13655](https://vajira.max.gov/browse/APPEALS-13655) | [APPEALS-13659](https://vajira.max.gov/browse/APPEALS-13659) |

### Description
Made changes to Gemfile to upgrade rails.
- rails updated to 5.2.8
- Additional changes are needed to keep specific gems to match original build according to the Gemfile.lock file prior to rails update.  gems listed below:
ddtrace - 0.34.1 (datadog)
redis-rails - 5.0.2 
database_cleaner - 1.7.0
debase-ruby_core_source - 0.10.14
knapsack_pro - 2.8.0

# M1 re-setup:
- run`export OCI_DIR=~/Downloads/instantclient_19_8`
- run `bundle install `to run new Gemfile. 
- when running bundle install pg gem might fail. you'll have to run script from m1 setup doc. 
     - the bundle install command will fail b/c the pg gem must be installed manually after the script executes. To install the pg gem manually run the following command: `gem install pg -v '1.1.4' -- --with-pg-config=/Applications/Postgres.app/Contents/Versions/latest/bin/pg_config`
- run `bundle install` to confirm all dependencies have been successfully updated

### Acceptance Criteria
- [x] Code compiles correctly
- [x] Test suite runs

### Testing Plan
1. Go to ...

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Post this PR in #appeals-schema with a summary
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
* [ ] Update Fakes
* [ ] Integrations impacting functionality are tested in Caseflow UAT

### CODE REVIEW

**Comment by mikefinneran** [comment]
Includes rails-html-sanitizer, loofah, nokogiri, miniportile, and racc which were also updated individually here https://github.com/department-of-veterans-affairs/caseflow/pull/18020

File: Gemfile.lock [hunk 1]
```diff
@@ -427,14 +427,22 @@ GEM
...
+    net-protocol (0.2.1)
+      timeout
+    net-smtp (0.3.3)
+      net-protocol
+    newrelic_rpm (8.15.0)
     nio4r (2.5.8)
     no_proxy_fix (0.1.2)
-    nokogiri (1.13.1)
-      mini_portile2 (~> 2.7.0)
+    nokogiri (1.14.1-x86_64-darwin)
```
Comment (mikefinneran) [inline Gemfile.lock:442]: Is this OS specific?

File: spec/feature/queue/motion_to_vacate_spec.rb [hunk 2]
```diff
@@ -897,8 +897,7 @@ def return_to_lit_support(disposition:)
           )
 
           expect(vacate_stream.decision_issues.size).to eq(6)
-
-          remanded = vacate_stream.decision_issues.select { |di| di.remanded? }
+          remanded = vacate_stream.decision_issues.select :remanded?
```
Comment (mikefinneran) [inline spec/feature/queue/motion_to_vacate_spec.rb:900]: Is that the same as?
```
remanded = vacate_stream.decision_issues.select(&:remanded?)
```

File: Gemfile.lock (hunk 1 above)
Comment (Jruuuu) [reply Gemfile.lock:442]: As I do my research. I believe this installs the Native Gems based on your OS supported Platforms

[Installing Nokogiri](https://nokogiri.org/tutorials/installing_nokogiri.html#installing-native-gems)

so yes this is OS specific, but when you run gem install or bundle install. It looks for the supported platforms and installs the specific lib

from nokogiri website:
Installing Native Gems[¶](https://nokogiri.org/tutorials/installing_nokogiri.html#installing-native-gems)
Faster, more reliable installation[¶](https://nokogiri.org/tutorials/installing_nokogiri.html#faster-more-reliable-installation)
"Native gems" contain pre-compiled libraries for a specific machine architecture. On supported platforms, this removes the need for compiling the C extension and the packaged libraries, or to install system dependencies. This results in much faster installation and more reliable installation, which as you probably know have historically been challenging topics for Nokogiri users.

If you're on a supported platform, either gem install or bundle install should install a native gem without any additional action on your part.

**Pull request closed without merging**